ogr/ogrsf_frmts/oci/ogrocidatasource.cpp

```diff
@@ -900,33 +920,56 @@ int OGROCIDataSource::FetchSRSId(OGRSpatialReference *poSRS)
 
     /* ==================================================================== */
     /*      The first strategy is to see if we can identify it by           */
-    /*      authority information within the SRS.  Either using ORACLE      */
-    /*      authority values directly, or check if there is a known         */
-    /*      translation for an EPSG authority code.                         */
+    /*      authority information within the SRS.                           */
     /* ==================================================================== */
     const char *pszAuthName = nullptr, *pszAuthCode = nullptr;
 
-    if (poSRS->IsGeographic())
-    {
-        pszAuthName = poSRS->GetAuthorityName("GEOGCS");
-        pszAuthCode = poSRS->GetAuthorityCode("GEOGCS");
-    }
-    else if (poSRS->IsProjected())
+    if (poSRS->IsGeographic() || poSRS->IsProjected())
     {
-        pszAuthName = poSRS->GetAuthorityName("PROJCS");
-        pszAuthCode = poSRS->GetAuthorityCode("PROJCS");
+        pszAuthName = poSRS->GetAuthorityName(nullptr);
+        pszAuthCode = poSRS->GetAuthorityCode(nullptr);
     }
 
+    OGROCIStatement oCmdStatement(GetSession());
+
     if (pszAuthName != nullptr && pszAuthCode != nullptr)
     {
+        // If the authority is Oracle, use the code as the SRID
         if (EQUAL(pszAuthName, "Oracle") && atoi(pszAuthCode) != 0)
             return atoi(pszAuthCode);
 
+        // If the authority is EPSG, try to retrieve the record of CS_SRS
+        // whose SRID is the EPSG code. And if the WKT definition from the
+        // CS_SRS record is equivalent to the one of poSRS, use the EPSG code
+        // as the SRID. This works for a number
```

Comment:
"This works for a number" is there a missing sentence closure ?

Reply:
indeed. completed as "This works for a number of EPSG CRS (4326, 3857, etc.)"